src/app/modules/auth/auth.middlewares.ts

```diff
@@ -92,3 +95,45 @@ export const validateVerifyOtpParams = celebrate({
       .message('Please enter a valid OTP'),
   }),
 })
+
+/**
+ * Middleware that only allows users with a valid bearer token to pass through to the next handler
+ */
+export const authenticateApiKey: ControllerHandler<
+  unknown,
+  unknown,
+  ApiReqBody
+> = (req, res, next) => {
+  const authorizationHeader = req.headers.authorization
+  if (!authorizationHeader) {
+    return res
+      .status(StatusCodes.UNAUTHORIZED)
+      .json({ message: 'Authorisation header is missing' })
+  }
+  const [bearerString, apiKey] = authorizationHeader.split(BEARER_SEPARATOR)
+  if (bearerString !== BEARER_STRING) {
+    return res
+      .status(StatusCodes.BAD_REQUEST)
+      .json({ message: 'Invalid authorisation header format' })
+  }
+  const [apiEnv, apiVersion, userId, token] = apiKey.split(API_KEY_SEPARATOR)
+  if (!apiEnv || !apiVersion || !userId || !token) {
+    return res
+      .status(StatusCodes.BAD_REQUEST)
+      .json({ message: 'Invalid API key format' })
+  }
+  return getUserByApiKey(userId, token)
+    .map((user) => {
+      if (!user) {
+        return res
+          .status(StatusCodes.UNAUTHORIZED)
+          .json({ message: 'Invalid API key' })
+      }
+      req.body.formSg = { userId: user.id }
```

Comment:
Same comment as earlier, the namespace `req.body` feels wrong. body means something, and represents what the request actually sent (as-in: request header; request body), if we have a standard place to save the authenticated user, let's use that (in the session object), and if not, we can create a completely separate namespace to indicate that this is custom metadata.

Reply:
(as above) [32c1cc1](https://github.com/opengovsg/FormSG/pull/6376/commits/32c1cc1935b89e34e5abe9fa3803ea6520cef0e7)